pkg/synccompactor/compactor.go

```diff
@@ -81,55 +86,35 @@ func (c *Compactor) Compact(ctx context.Context) (*CompactableSync, error) {
 
 		compactable, err := c.doOneCompaction(ctx, base, applied)
 		if err != nil {
-			if err := removeIntermediateFiles(intermediates, false); err != nil {
-				return nil, err
-			}
 			return nil, err
 		}
 		// Collect all the intermediate files we create to remove at the end
 		intermediates = append(intermediates, compactable.FilePath)
 		base = compactable
 	}
 
-	if len(intermediates) > 0 {
-		if err := removeIntermediateFiles(intermediates, true); err != nil {
-			return nil, err
-		}
-	}
-
 	// Move last compacted file to the destination dir
 	finalPath := path.Join(c.destDir, fmt.Sprintf("compacted-%s.c1z", base.SyncID))
-	// Attempt to move via rename
-	if err := os.Rename(base.FilePath, finalPath); err != nil {
-		var linkErr *os.LinkError
-		if errors.As(err, &linkErr) {
-			// Mac err table: https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/intro.2.html
-			// Win err table: https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/18d8fbe8-a967-4f1c-ae50-99ca8e491d2d?redirectedfrom=MSDN
-			if errors.Is(linkErr.Err, syscall.Errno(0x12)) || (runtime.GOOS == "windows" && errors.Is(linkErr.Err, syscall.Errno(0x11))) {
-				// If rename doesn't work do a full create/copy
-				if err := mvFile(base.FilePath, finalPath); err != nil {
-					// Return if mv file failed
-					return nil, err
-				}
-			} else {
-				// Return if it's a different kind of link err
-				return nil, err
-			}
-		} else {
-			// Return if it's not a link err
+	if err := mvFile(base.FilePath, finalPath); err != nil {
+		return nil, err
+	}
+
+	if !filepath.IsAbs(finalPath) {
+		abs, err := filepath.Abs(finalPath)
+		if err != nil {
 			return nil, err
 		}
+		finalPath = abs
 	}
-	base.FilePath = finalPath
-
-	return base, nil
+	return &CompactableSync{FilePath: finalPath, SyncID: base.SyncID}, nil
 }
 
 func mvFile(sourcePath string, destPath string) error {
 	source, err := os.Open(sourcePath)
 	if err != nil {
 		return fmt.Errorf("failed to open source file: %w", err)
 	}
+	defer source.Close()
```

Comment:
This now copies, not moves, right?

Reply:
Correct, that's why moving the close from defer to here is ok, because we copy it and leave the original for the caller of the compactor to cleanup later.

Reply:
Oh yes good point